.github/workflows/required_lint_check.yaml

```diff
@@ -0,0 +1,110 @@
+#
+# For only the optional and more verbose feedback, use the 
+# `optional_lint_check.yaml`.
+#
+# The required lint check, is, well, required to pass and WILL block merges and builds,
+# but is much more opinionated and slim (there are only a few things we error on).
+#
+# The optional lint check runs on all PRs to provide info
+# but doesn't actually block any merge or build
+#
+on: 
+  workflow_call:
+    inputs:
+      python-version:
+        description: 'The Python version to use and defaults to latest'
+        required: false
+        default: '3.9'
+        type: string
+      use-cache:
+        required: false
+        default: true
+        type: boolean
+jobs:
+  build:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ inputs.python-version }}
+
+      - name: Checkout Code
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+
+      - uses: actions/download-artifact@v3
+        with:
+          # this is uploaded by the lint-create-config.yaml workflow
+          name: linters-config
+          path: .github/linters
+
+      - name: Display files
+        run: ls -aR 
+
+      # cache the dependencies based on a hash of the lock file
+      - name: Poetry cache
+        if: inputs.use-cache
+        id: cache-poetry
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pypoetry/virtualenvs
+          key: ${{ runner.os }}-poetry-${{ hashFiles('**/poetry.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-poetry-
+
+      - name: Install poetry so virtual env has imports that pylint can check
+        # that is poetry's recommended way to install
+        run: |
+          pip install --upgrade pip
+          curl -sSL https://install.python-poetry.org | python -
+          pip list
+          poetry install -vv --all-extras --no-interaction
+          poetry show
+
+      - name: Save poetry cache if cache miss
+        if: inputs.use-cache && steps.cache-poetry.outputs.cache-hit != 'true'
+        run: echo "Cache miss, saving the cache"
+
+      - name: Use Python module path to update necessary linter configs
+        run: |
+          virtual_env=$(poetry env info --path)
+          echo "Python Virtual Env Path: $virtual_env"
+
+          site_packages_folder=$(find $virtual_env -type d -name "site-packages" -print -quit)
+          if [ -n "$site_packages_folder" ]; then
+            echo "Found 'site-packages' folder at: $site_packages_folder"
+          else
+            echo "No 'site-packages' folder found in provided virtual env: $virtual_env."
+            exit 1
+          fi
+
+          echo
+          echo "Updating .github/linters/.python-lint"
+          echo "init-hook='import sys; sys.path.append(\".\"); sys.path.append(\"/tmp/lint\"); sys.path.append(\"$site_packages_folder\")'"
```

Comment:
Is this echo statement used for debugging/info? It looks like it is replicating line 93.

Reply:
Yeah it's for debugging what actually gets written into the .python-lint file in the logs